content/common/utils/utils.h

```diff
@@ -0,0 +1,49 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+
+#ifndef UTILS_H_
+#define UTILS_H_ 1
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#if defined(__linux__) || defined(__Unikraft__)
```

Comment:
This should be compiler specific, I think, not OS-specific. So mostly something like:
```
#if defined(__GNUC__)
...
#elif defined(__MSVC__)
...
#endif
```

Reply:
How is this helpful? The difference between the two error-handling mechanisms here is the platform. Yes these platforms also use different compilers, but this is a by-product. The Linux code can also be compiled with clang, for example.

Reply:
I would include macOS and *BSD macros also. I can help test it on macOS. I think you could help with macOS and others with *BSD.